### ISSUE TITLE
Modules within modules are attempted to be loaded on the wrong relative path

### ISSUE THREAD

**Issue by ksommoxy** [aquasecurity/tfsec#971]
**Describe the bug**

When using nested modules/sub-modules, the relative path of a child module within a parent module is based on the path of the consumer of the parent module instead of relative to the parent module's location, causing the module load to fail with a warning.

**To Reproduce**

With a sample configuration/set of modules:
`repo/configurations/project/main.tf`
```
module "storage" {
    source     = "../../modules/azure/storage-account"
    name       = "mystorageaccount"
    containers = { "mycontainer" = {} }
}
```

`repo/modules/azure/storage-account/module.tf`
```
resource "azurerm_storage_account" "storage_accounts" {
    name = var.name
}

module "storage_container" {
    for_each = var.containers
    source   = "./../storage-container"
    storage_account_name = resource.azurerm_storage_account.storage_accounts.name
}
```

`repo/modules/azure/storage-container/module.tf`
```
resource "azurerm_storage_container" "containers" {
    name                  = var.name
    storage_account_name  = var.storage_account_name
}
```

Then running the following steps within the `repo/configurations/project/` path:
1. `terraform init` (successfully initialized)
2. `tfsec . -f json`, which results in:

```
WARNING: Failed to load module: failed to load module storage_container["mycontainer"]: 
open C:\...\repo\configurations\storage-container: The system cannot find the file specified.
{
        "results": null
}
```

TFSec is attempting to load the child module (storage-container) in the path `repo/configurations/storage-container` (relative to the working directory TFSec was ran) instead of relative to the module/TF that is referencing it (storage-account).

**Expected behavior**

Expected that the child module will be loaded succesfully.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Terraform: 1.0.3
 - TFSec: 0.51.0

**Additional context**

**Comment by owenrumney** [comment]
Hey @ksommoxy I've tried to reproduce this in https://github.com/aquasecurity/tfsec/pull/976/files as you've described and I can't reproduce. I added some variables to make it validate, below is the output I get. Can you check the PR above to see if anything is different to yours to help me debug this

```
~/c/a/tfsec/e/9/configurations/project $ tfsec --version
0.51.0

~/c/a/tfsec/e/9/configurations/project $ terraform --version                                                       
Terraform v1.0.3
on linux_amd64

~/c/a/tfsec/e/9/configurations/project $ tfsec . -f json 

{
	"results": [
		{
			"rule_id": "AZU015",
			"rule_description": "The minimum TLS version for Storage Accounts should be TLS1_2",
			"rule_provider": "azure",
			"impact": "The TLS version being outdated and has known vulnerabilities",
			"resolution": "Use a more recent TLS/SSL policy for the load balancer",
			"links": [
				"https://tfsec.dev/docs/azure/AZU015/",
				"https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#min_tls_version",
				"https://docs.microsoft.com/en-us/azure/storage/common/transport-layer-security-configure-minimum-version"
			],
			"location": {
				"filename": "/home/owenaqua/code/aquasecurity/tfsec/example/971/modules/azure/storage-account/module.tf",
				"start_line": 1,
				"end_line": 4
			},
			"description": "Resource 'module.storage:azurerm_storage_account.storage_accounts' should have the min tls version set to TLS1_2 .",
			"severity": "CRITICAL",
			"status": "failed"
		}
	]
}

```

**Comment by owenrumney** [comment]
I can reproduce this when I remove .terraform init, we're looking into this now

**Comment by owenrumney** [comment]
Hey @ksommoxy - this is fixed in v0.53.3
